File: go-sandbox/tests/strs/strs.go
// Package strs contains tiny string helpers used by the test suite.
package strs

import (
	"strings"
	"unicode"
)

func Reverse(s string) string {
	r := []rune(s)
	for i, j := 0, len(r)-1; i < j; i, j = i+1, j-1 {
		r[i], r[j] = r[j], r[i]
	}
	return string(r)
}

func IsPalindrome(s string) bool {
	var cleaned []rune
	for _, r := range s {
		if unicode.IsLetter(r) || unicode.IsDigit(r) {
			cleaned = append(cleaned, unicode.ToLower(r))
		}
	}
	for i, j := 0, len(cleaned)-1; i < j; i, j = i+1, j-1 {
		if cleaned[i] != cleaned[j] {
			return false
		}
	}
	return true
}

func WordCount(s string) int {
	return len(strings.Fields(s))
}

func Shout(s string) string {
	return strings.ToUpper(s)
}

func Capitalize(s string) string {
	if s == "" {
		return s
	}
	r := []rune(s)
	r[0] = unicode.ToUpper(r[0])
	return string(r)
}

File: lua-sandbox/test-busted/src/sandbox/strs.lua
-- String helpers.
local M = {}

function M.reverse(s) return s:reverse() end

function M.is_palindrome(s)
  local cleaned = s:lower():gsub("[^%w]", "")
  return cleaned == cleaned:reverse()
end

function M.word_count(s)
  local n = 0
  for _ in s:gmatch("%S+") do n = n + 1 end
  return n
end

function M.capitalize(s)
  if s == "" then return s end
  return s:sub(1, 1):upper() .. s:sub(2)
end

function M.shout(s) return s:upper() end

return M
